### PR TITLE
Upgraded searchkit to latest beta

### LINKED ISSUE #2151: Typo in no hits copy
screenshot from 2016-12-16 15-21-07

This typo is located in searchkit so we'll have to replace that component with our own

### LINKED ISSUE #2737: Upgrade to latest searchkit beta

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Fixes #2737 
Fixes #2151 

#### What's this PR do?
Upgrades the version of searchkit used and removes some workarounds we don't need anymore

#### How should this be manually tested?
Go to `/learners` and try out every feature on the page. It should look and act just like it did before

### CODE REVIEW

File: js_test.sh [hunk 1]
```diff
@@ -32,13 +32,13 @@ fi
 if [[ $(
     cat "$TMP_FILE" |
     grep -v 'ignored, nothing could be mapped' |
-    grep -v 'You are manually calling a React.PropTypes validation function' |
-    grep -v 'React.__spread is deprecated' |
```
Comment (aliceriot) [inline js_test.sh]: yay!

File: package.json [hunk 2]
```diff
@@ -106,7 +106,7 @@
     "sanctuary": "^0.11.1",
     "sass-lint": "^1.10.2",
     "sass-loader": "^4.1.1",
-    "searchkit": "0.10.0",
+    "searchkit": "rc",
```
Comment (aliceriot) [inline package.json]: Could be make sure the version we're installing is reflected here?

File: static/js/babelhook.js [hunk 3]
```diff
@@ -2,8 +2,3 @@ const { babelSharedLoader } = require("../../webpack.config.shared");
 require('babel-polyfill');
 
 require('babel-register')(babelSharedLoader.query);
-
-// Force everything to the latest version of react
-let React = require('react');
-let mockRequire = require('mock-require');
```
Comment (aliceriot) [inline static/js/babelhook.js]: are we using `mock-require` anywhere else? if not it might be good to remove the dependency.

Comment (noisecapella) [inline static/js/babelhook.js]: No, I added it for this purpose. I'll remove it

File: package.json (hunk 2 above)
Comment (noisecapella) [inline package.json]: Sure

**Comment by noisecapella** [comment]
Comments addressed, and I also removed searchkit-multiselect since we are using a modified version of it anyway

**Pull request merged**